rplugin/python3/deoplete/source/buffer.py

```diff
@@ -29,8 +29,12 @@ def on_event(self, context):
     def gather_candidates(self, context):
         self.__make_cache(context)
 
-        buffers = [x['candidates'] for x in self.__buffers.values()
-                   if x['filetype'] in context['filetypes']]
+        buffer_values = self.__buffers.values()
+        if context['vars'].get('require_same_filetype', True):
+          buffer_values = [x for x in buffer_values
```

Comment:
Oops, I think it can just be:

```python
if not same_filetype or x['filetype'] in context['filetypes']
```
instead of

```python
if not same_filetype or (same_filetype and x['filetype'] in context['filetypes'])
```

Right?  I think I need to get some sleep.